src/QMCWaveFunctions/Fermion/DiracMatrixComputeCUDA.hpp

```diff
@@ -227,25 +236,27 @@ class DiracMatrixComputeCUDA : public Resource
    *  There is no optimization (yet) for TMAT same type as TREAL
    */
   template<typename TMAT>
-  void invert_transpose(compute::BLASHandle<PlatformKind::CUDA>& cuda_handles,
+  void invert_transpose(compute::Queue<PlatformKind::CUDA>& queue,
                         DualMatrix<TMAT>& a_mat,
                         DualMatrix<TMAT>& inv_a_mat,
                         DualVector<LogValue>& log_values)
   {
+    cudaStream_t h_stream = queue.getNative();
+    cublasErrorCheck(cublasSetStream(h_cublas_, h_stream), "cublasSetStream failed!");
```

Comment:
We don't have this scenario.

Reply:
In terms of stream/cublas association. It can still go wrong with the old code if you construct two `compute::BLASHandle` with the same `compute::Queue`

Reply:
This function and the the mw function have according to AMD documentation a requirement that it always be called on the same queue or synchronize has been called on that last queue since the inverter was last called. Unless `hipblasSetStream` contains code that calls a stream sync on the last stream when this is called and it does not appear to. We should document this requirement on the queue argument.  If for instance this object migrated between threads in the future this could be quite a surprise.

When I wrote or reviewed the old code I was not really aware of all the ways hip API's could be different than AMD ones.

Reply:
This is what I found in the [implementation](https://github.com/ROCm/hipBLAS/blob/develop/library/src/amd_detail/hipblas.cpp). The file was too large and cannot provide a link.
```
hipblasStatus_t hipblasSetStream(hipblasHandle_t handle, hipStream_t streamId)
try
{
    if(handle == nullptr)
    {
        return HIPBLAS_STATUS_NOT_INITIALIZED;
    }
    return hipblasConvertStatus(rocblas_set_stream((rocblas_handle)handle, streamId));
}
catch(...)
{
    return hipblas_exception_to_status();
}
```

Reply:
Added a comment at the top of the class
```
*  All the public APIs are synchronous. The asynchronous queue argument gets synchronized before return.
*  rocBLAS, indirectly used via hipBLAS, requires synchronizing the old stream before setting a new one.
*  We don't need to actively synchronize the old stream because it gets synchronized right after each use.
```

Reply:
I missed this... I guess this makes us a more safe but is not what I would expect at all.